src/OpenTelemetry.Extensions.Enrichment/README.md

```diff
@@ -6,6 +6,87 @@
 Contains OpenTelemetry .NET SDK telemetry enrichment framework
 which is used for enrichment of traces.
 
+## Introduction
+
+Telemetry enrichment attaches various types of information to traces.
+Examples of the types of information that can be attached are environment
+information, machine information, and user details.
+
+With the enrichment framework, you don't need to worry about attaching
+the information carefully to each telemetry object you touch.
+Instead, if you implement your enricher class, it  takes care of the details
+automatically. You simply register your class with the enrichment framework
+and the enrichment framework will make sure to call the `Enrich()` method of your
+class every time there is an `Activity` in your app.
+
 ## Traces
 
-TBD
+Currently this package supports trace enrichment only.
+
+### Installation
+
+Download the `OpenTelemetry.Extensions.Enrichment` package:
+
+```shell
+dotnet add package OpenTelemetry.Extensions.Enrichment
+```
+
+### Usage
+
+Add your custom trace enricher:
+
+```csharp
+using OpenTelemetry.Extensions.Enrichment;
+using OpenTelemetry.Trace;
+using Microsoft.Extensions.DependencyInjection;
+
+public void ConfigureServices(IServiceCollection services)
```

Comment:
what is this method? This example is missing context..

Reply:
Updated everything as per the change requested